src/TSTransformer/util/arrayBindingPatternContainsHoists.ts

```diff
@@ -0,0 +1,21 @@
+import { TransformState } from "TSTransformer/classes/TransformState";
+import { checkVariableHoist } from "TSTransformer/util/checkVariableHoist";
+import ts from "typescript";
+
+export function arrayBindingPatternContainsHoists(
+	state: TransformState,
+	arrayBindingPattern: ts.ArrayBindingPattern,
+): boolean {
+	for (const element of arrayBindingPattern.elements) {
+		if (ts.isBindingElement(element) && ts.isIdentifier(element.name)) {
+			const symbol = state.typeChecker.getSymbolAtLocation(element.name);
+			if (symbol) {
+				checkVariableHoist(state, element.name, symbol);
+				if (state.isHoisted.get(symbol)) {
+					return true;
+				}
+			}
+		}
+	}
+	return false;
```

Comment:
Is this safe to default to false? What about the elements that are not `bindingElement` as checked inside the loop?

Reply:
If it's not a `ts.BindingElement`, it must be a `ts.OmittedExpression`. In that case, doesn't create a variable.
For non-identifiers, it must be a nested array/object binding pattern. For those cases, the hoisting logic is handled elsewhere and the variable here will be a tempId.

Reply:
This should go in a comment in the file, maybe?